File: src/com/adventofcode/december12/MainProgram.java
package com.adventofcode.december12;

public class MainProgram {

    public static void main(String[] args) {
        MainProgram pp = new MainProgram();
        long output = pp.run();
        System.out.println("Puzzle output : " + output);
    }

    //-----------------------------------------------------------

    private Graph graph;
    private static final String monthNumber = "12";
    private static final String postfix = "1";

    public MainProgram() {
        Input input = new Input("input" + monthNumber + "_" + postfix);
        graph = input.getGraph();
        System.out.println();
    }

    public long run() {
        long sum = graph.findPaths();
        return sum;
    }
}

File: src/com/adventofcode/december01/MainProgram.kt
package com.adventofcode.december01

import com.adventofcode.general.Input
import com.adventofcode.general.getDayOfMonthFromClassName

fun main() {
    MainProgram().run()
}

class MainProgram {
    private val test = false

    fun run() {
        val dayOfMonth = getDayOfMonthFromClassName(this)
        val input = Input(test, dayOfMonth)

        val puzzleSolution = PuzzleSolution(input)
        val output1 = puzzleSolution.resultPartOne()
        val output2 = puzzleSolution.resultPartTwo()

        println("December $dayOfMonth")
        println("  Puzzle output part 1: $output1")
        println("  Puzzle output part 2: $output2")
    }
}